programs/clearing_house/src/controller/orders.rs

```diff
@@ -453,12 +460,19 @@ pub fn fill_order(
     let oracle_price: i128;
     {
         let market = &mut market_map.get_ref_mut(&market_index)?;
-        validate!(
-            ((oracle_map.slot == market.amm.last_update_slot && market.amm.last_oracle_valid)
-                || market.amm.curve_update_intensity == 0),
-            ErrorCode::AMMNotUpdatedInSameSlot,
-            "AMM must be updated in a prior instruction within same slot"
-        )?;
+
+        if market.expiry_ts != 0 {
```

Comment:
why are you only calling this in fill? seems like it should be in a controller and called in more places?

Reply:
added to `update_amm` as well